commit 16b389e00a828cfc114cfc0bcaa64d9e247fd890
Author: Dan Williams <dcbw@redhat.com>
Date:   Thu Jul 9 13:37:50 2009 -0400

    eap: fix file chooser button validation
    
    file-set only emits on /user/ initiated events, not for example when
    the file chooser has finally completed the async file operations after
    gtk_file_chooser_set_filename().

diff --git a/network-manager-applet-0.7.1/src/wireless-security/eap-method-peap.c b/network-manager-applet-0.7.1/src/wireless-security/eap-method-peap.c
index 8d82fee..d9f4f46 100644
--- a/network-manager-applet-0.7.1/src/wireless-security/eap-method-peap.c
+++ b/network-manager-applet-0.7.1/src/wireless-security/eap-method-peap.c
@@ -309,7 +309,7 @@ eap_method_peap_new (const char *glade_file,
 	gtk_file_chooser_set_local_only (GTK_FILE_CHOOSER (widget), TRUE);
 	gtk_file_chooser_button_set_title (GTK_FILE_CHOOSER_BUTTON (widget),
 	                                   _("Choose a Certificate Authority certificate..."));
-	g_signal_connect (G_OBJECT (widget), "file-set",
+	g_signal_connect (G_OBJECT (widget), "selection-changed",
 	                  (GCallback) wireless_security_changed_cb,
 	                  parent);
 	filter = eap_method_default_file_chooser_filter_new (FALSE);
diff --git a/network-manager-applet-0.7.1/src/wireless-security/eap-method-tls.c b/network-manager-applet-0.7.1/src/wireless-security/eap-method-tls.c
index 7d78d07..6912427 100644
--- a/network-manager-applet-0.7.1/src/wireless-security/eap-method-tls.c
+++ b/network-manager-applet-0.7.1/src/wireless-security/eap-method-tls.c
@@ -311,14 +311,14 @@ setup_filepicker (GladeXML *xml,
 	 * and desensitize the user cert button.
 	 */
 	if (privkey) {
-		g_signal_connect (G_OBJECT (widget), "file-set",
+		g_signal_connect (G_OBJECT (widget), "selection-changed",
 		                  (GCallback) private_key_picker_file_set_cb,
 		                  method);
 		if (filename)
 			private_key_picker_helper ((EAPMethod *) method, filename, FALSE);
 	}
 
-	g_signal_connect (G_OBJECT (widget), "file-set",
+	g_signal_connect (G_OBJECT (widget), "selection-changed",
 	                  (GCallback) wireless_security_changed_cb,
 	                  parent);
 
diff --git a/network-manager-applet-0.7.1/src/wireless-security/eap-method-ttls.c b/network-manager-applet-0.7.1/src/wireless-security/eap-method-ttls.c
index b455fc8..78d0f80 100644
--- a/network-manager-applet-0.7.1/src/wireless-security/eap-method-ttls.c
+++ b/network-manager-applet-0.7.1/src/wireless-security/eap-method-ttls.c
@@ -341,7 +341,7 @@ eap_method_ttls_new (const char *glade_file,
 	gtk_file_chooser_set_local_only (GTK_FILE_CHOOSER (widget), TRUE);
 	gtk_file_chooser_button_set_title (GTK_FILE_CHOOSER_BUTTON (widget),
 	                                   _("Choose a Certificate Authority certificate..."));
-	g_signal_connect (G_OBJECT (widget), "file-set",
+	g_signal_connect (G_OBJECT (widget), "selection-changed",
 	                  (GCallback) wireless_security_changed_cb,
 	                  parent);
 	filter = eap_method_default_file_chooser_filter_new (FALSE);
